src/python_testing/TC_CGEN_2_2.py

```diff
@@ -0,0 +1,796 @@
+        th1_certs_new = await self.default_controller.IssueNOCChain(csr_update, self.dut_node_id)
+        new_root_cert = th1_certs_new.rcacBytes
+
+        # Flow generates a new TrustedRootCertificate - Send command to add new trusted root certificate
+        cmd = cluster_opcreds.Commands.AddTrustedRootCertificate(new_root_cert)
+        resp = await self.send_single_cmd(dev_ctrl=self.default_controller, node_id=self.dut_node_id, cmd=cmd)
+
+        self.step(31)
+        trusted_root_list_original_updated = await self.read_single_attribute_check_success(
+            dev_ctrl=self.default_controller,
+            node_id=self.dut_node_id,
+            cluster=cluster_opcreds,
+            attribute=cluster_opcreds.Attributes.TrustedRootCertificates)
+        trusted_root_list_original_size_updated = len(trusted_root_list_original_updated)
+        logger.info(f'Step #31 - The updated num_trusted_roots_original: {trusted_root_list_original_size_updated}')
+        # Verify that the trusted root list size has increased by 1
+        asserts.assert_equal(trusted_root_list_original_size_updated, trusted_root_list_original_size + 2,
+                             "Step #31 - Unexpected number of entries in the TrustedRootCertificates table after update")
+
+        self.step(32)
+        cmd = Clusters.GeneralCommissioning.Commands.ArmFailSafe(expiryLengthSeconds=maxFailsafe)
+        resp = await self.send_single_cmd(
+            dev_ctrl=self.default_controller,
+            node_id=self.dut_node_id,
+            cmd=cmd)
+        # Verify that the DUT responds with ArmFailSafeResponse with ErrorCode as 'OK'(0)
+        asserts.assert_equal(resp.errorCode, Clusters.GeneralCommissioning.Enums.CommissioningErrorEnum.kOk,
+                             "Failure status returned from arm failsafe")
+        # Verify that DebugText is empty or has a maximum length of 512 characters
+        debug_text = resp.debugText
+        assert debug_text == '' or len(debug_text) <= 512, "debugText must be empty or have a maximum length of 512 characters"
+        logger.info(f'Step #32 - ArmFailSafeResponse with ErrorCode as OK({resp.errorCode})')
+
+        self.step(33)
+        # Step 33 - Bypassing the wait for the failsafe timer to expire (PIXIT.CGEN.FailsafeExpiryLengthSeconds)
+        # Originally, this step required waiting for the failsafe timer to expire after a specific duration defined in PIXIT.CGEN.FailsafeExpiryLengthSeconds.
+        # However, in CI environments, this wait introduces unnecessary delays. To avoid this, we use the `expire_failsafe_timer` function,
+        # which immediately expires the failsafe timer (sets it to 0), bypassing the natural wait and speeding up the test execution.
+
+        # This function bypasses the wait for the FailsafeTimer to expire for TH2 as originally defined in the test plan.
+        resp = await self.expire_failsafe_timer(dev_ctrl=TH2, node_id=newNodeId+1)
```

Comment:
So the point of step 33 was to ensure that the timer DOESN'T expire at the original set time, but was instead updated by the second arm failsafe call. So this shouldn't be here, but also the assert below should be failing if this failsafe has been expired